src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/processaudiovideo/ProcessAudioVideoEvidenceAboutToSubmitHandler.java

```diff
@@ -103,27 +107,30 @@ private void processIfIncludeEvidence(SscsCaseData caseData, PreSubmitCallbackRe
             List<DwpDocument> dwpDocuments = new ArrayList<>();
 
             for (AudioVideoEvidence audioVideoEvidence : caseData.getAudioVideoEvidence()) {
+                if (isSelectedEvidence(audioVideoEvidence, caseData)) {
 
-                if (UploadParty.DWP.equals(audioVideoEvidence.getValue().getPartyUploaded())) {
-                    dwpDocuments.add(buildAudioVideoDwpDocument(audioVideoEvidence, response));
-                } else {
-                    sscsDocuments.add(buildAudioVideoSscsDocument(audioVideoEvidence, response));
-                }
-            }
+                    if (UploadParty.DWP.equals(audioVideoEvidence.getValue().getPartyUploaded())) {
+                        dwpDocuments.add(buildAudioVideoDwpDocument(audioVideoEvidence, response));
```

Comment:
Remove the buildAudioVideoDwpDocument method. You won't need to build it anymore as you are already building it as part of the midevent callback.

Reply:
also you can use DocumentUtil util class for file extention check, there are couple of methods already.